syntax = "proto3";

import "EchoAttributes.proto";

package testing;

message GpioOut {
  bool value = 1;
}

message GpioIn {
  bool value = 1;
  bool valueN = 2;
}

message GpioState
{
    bool state = 1;
    uint32 pin = 2;
}

service Command
{
    option (service_id) = 1;

    rpc RunGpioTest(GpioOut) returns (Nothing) { option (method_id) = 1; }
}

service Reponse
{
    option (service_id) = 2;

    rpc GpioTestResult(GpioIn) returns (Nothing) { option (method_id) = 1; }
}

service GpioTester
{
    option (service_id) = 3;

    rpc SetGpio(GpioState) returns (Nothing) { option (method_id) = 1; }
}

service GpioTested
{
    option (service_id) = 4;

    rpc SetGpio(GpioState) returns (Nothing) { option (method_id) = 1; }
}

service GpioObserver
{
    option (service_id) = 5;

    rpc TesterGpioChanged(GpioState) returns (Nothing) { option (method_id) = 1; }
    rpc TestedGpioChanged(GpioState) returns (Nothing) { option (method_id) = 2; }
}
